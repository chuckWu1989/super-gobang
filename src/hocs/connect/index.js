import React from 'react';

export const connect = method => (
  WrappedComponent => (
    (props) => {
      const methodWithProps = typeof method === 'function' ? method(props) : {};
      return (
        <WrappedComponent {...methodWithProps} {...props} />
      );
    }
  )
);

export default connect;
