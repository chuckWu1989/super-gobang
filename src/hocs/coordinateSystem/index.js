import React, { Component } from 'react';
import { setCoordinate } from '../../controllers/coordinateSystem';

const coordinateSystem = WrappedComponent => (
  class extends Component {
    static defaultProps = {
      grid: 15,
      ratio: 1,
      width: 400,
    };
    componentWillMount() {
      setCoordinate.call(this);
    }
    componentWillReceiveProps() {
      setCoordinate.call(this);
    }
    render() {
      const { coord, width, height } = this.state;
      return coord ?
        (
          <WrappedComponent
            {...this.props}
            coord={this.state.coord}
            width={width}
            height={height}
          />
        ) :
        undefined;
    }
  }
);

export default coordinateSystem;
