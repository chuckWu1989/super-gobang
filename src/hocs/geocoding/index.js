import React, { Component } from 'react';
import { setCoordinate } from '../../controllers/geocoding';

const geocoding = WrappedComponent => (
  class extends Component {
    static defaultProps = {
      grid: 15,
      ratio: 1,
      width: 400,
    };
    componentWillMount() {
      setCoordinate.call(this);
    }
    componentWillReceiveProps() {
      setCoordinate.call(this);
    }
    render() {
      const {
        coord,
        width,
        height,
        unitWidth,
        unitHeight,
      } = this.state;
      return coord ?
        (
          <WrappedComponent
            {...this.props}
            coord={coord}
            width={width}
            height={height}
            unitWidth={unitWidth}
            unitHeight={unitHeight}
          />
        ) :
        undefined;
    }
  }
);

export default geocoding;
