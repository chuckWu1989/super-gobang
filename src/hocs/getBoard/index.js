import React from 'react';

const getBoard = (Component) => {
  const boardProps = {
    grid: 15,
    width:
  };
  boardProps.space = Math.floor()
  return (props) => <Component {...boardProps} {...props} />;
};

export default getBoard;
