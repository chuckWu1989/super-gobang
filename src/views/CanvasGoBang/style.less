.canvas-gobang-style {
  width: 100%;
  position: relative;
}
