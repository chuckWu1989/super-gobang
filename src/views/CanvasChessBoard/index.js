import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { drawBoard } from '../../controllers/CanvasChessBoard';
import './style.less';

class CanvasChessBoard extends Component {
  static propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }
  componentDidMount() {
    drawBoard(this.chessboard, this.props);
  }
  componentDidUpdate() {
    drawBoard(this.chessboard, this.props);
  }
  render() {
    const { width, height } = this.props;
    return (
      <canvas
        width={width}
        height={height}
        ref={(self) => { this.chessboard = self; }}
        className="canvas-chessboard-style"
      />
    );
  }
}

export default CanvasChessBoard;
