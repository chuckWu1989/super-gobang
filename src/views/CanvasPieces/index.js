import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.less';

class CanvasPieces extends Component {
  static defaultProps = {
    detectChessMovesChange: () => {},
    detectWidthChange: () => {},
  }
  static propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    detectChessMovesChange: PropTypes.func,
    detectWidthChange: PropTypes.func,
  }
  componentWillReceiveProps(nextProps) {
    this.props.detectChessMovesChange(this.canvas, nextProps);
  }
  componentDidUpdate(prevProps) {
    this.props.detectWidthChange(this.canvas, prevProps);
  }
  render() {
    const { width, height } = this.props;
    return (
      <canvas
        target
        ref={(self) => { this.canvas = self; }}
        width={width}
        height={height}
        className="canvas-pieces-style"
      />
    );
  }
}

export default CanvasPieces;
