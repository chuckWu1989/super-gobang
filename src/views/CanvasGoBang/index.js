import React from 'react';
import CanvasChessBoard from '../../router/CanvasChessBoard';
import CanvasPieces from '../../router/CanvasPieces';
import geocoding from '../../router/geocoding';
import './style.less';

export const CanvasGoBang = props => (
  <div className="canvas-gobang-style">
    <CanvasChessBoard {...props} />
    <CanvasPieces {...props} />
  </div>
);

export default geocoding(CanvasGoBang);
