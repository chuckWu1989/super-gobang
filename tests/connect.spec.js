import React from 'react';
import toJson from 'enzyme-to-json';
import { connect } from '../src/hocs/connect';
import { CanvasGoBang } from '../src/views/CanvasGoBang';

const { shallow } = global;

describe('connect', () => {
  describe('view', () => {
    it('should be rendered with method as props if method is a function', () => {
      const method = () => ({ test: () => {} });
      const WrappedComponent = connect(method)(CanvasGoBang);
      const tree = toJson(shallow(<WrappedComponent />));
      expect(tree).toMatchSnapshot();
    });
    it('should be rendered properly without method if method is not a function', () => {
      const WrappedComponent = connect(null)(CanvasGoBang);
      const tree = toJson(shallow(<WrappedComponent />));
      expect(tree).toMatchSnapshot();
    });
  });
});
