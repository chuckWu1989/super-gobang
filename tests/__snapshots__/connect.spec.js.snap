// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connect view should be rendered properly without method if method is not a function 1`] = `<CanvasGoBang />`;

exports[`connect view should be rendered with method as props if method is a function 1`] = `
<CanvasGoBang
  test={[Function]}
/>
`;
